sheet_test.go

```diff
@@ -49,6 +49,53 @@ func (s *SheetSuite) TestMakeXLSXSheetFromRows(c *C) {
 	c.Assert(xSI.T, Equals, "A cell!")
 }
 
+// Test if the NumFmts assigned properly according the FormatCode in cell.
+func (s *SheetSuite) TestMakeXLSXSheetWithNumFormats(c *C) {
+	file := NewFile()
+	sheet := file.AddSheet("Sheet1")
+	row := sheet.AddRow()
+
+	cell1 := row.AddCell()
+	cell1.Value = "A cell!"
+	cell1.numFmt = "general"
+
+	cell2 := row.AddCell()
+	cell2.Value = "37947.7500001"
+	cell2.numFmt = "0"
+
+	cell3 := row.AddCell()
+	cell3.Value = "37947.7500001"
+	cell3.numFmt = "mm-dd-yy"
+
+	cell4 := row.AddCell()
+	cell4.Value = "37947.7500001"
+	cell4.numFmt = "hh:mm:ss"
+
+	refTable := NewSharedStringRefTable()
+	styles := newXlsxStyleSheet(nil)
+	worksheet := sheet.makeXLSXSheet(refTable, styles)
+
+	c.Assert(styles.CellStyleXfs.Count, Equals, 1)
+	// The 0th CellStyleXf could just be getting the zero values by default
+	c.Assert(styles.CellStyleXfs.Xf[0].FontId, Equals, 0)
+	c.Assert(styles.CellStyleXfs.Xf[0].FillId, Equals, 0)
+	c.Assert(styles.CellStyleXfs.Xf[0].BorderId, Equals, 0)
+
+	c.Assert(styles.CellXfs.Count, Equals, 4)
+	c.Assert(styles.CellXfs.Xf[0].NumFmtId, Equals, 0)
+	c.Assert(styles.CellXfs.Xf[1].NumFmtId, Equals, 1)
+	c.Assert(styles.CellXfs.Xf[2].NumFmtId, Equals, 14)
+	c.Assert(styles.CellXfs.Xf[3].NumFmtId, Equals, 164)
+	c.Assert(styles.NumFmts.Count, Equals, 1)
+	c.Assert(styles.NumFmts.NumFmt[0].NumFmtId, Equals, 164)
+	c.Assert(styles.NumFmts.NumFmt[0].FormatCode, Equals, "hh:mm:ss")
+
```

Comment:
It seems to me that these tests should also make some assertions about the `FormattedValue()` of the cells. Does that make sense here?